test/test_matmul_cuda.py

```diff
@@ -458,6 +458,7 @@ def test_grouped_gemm_3d_2d(self, strided, a_row_major, b_row_major):
     def test_mm_bmm_dtype_overload(self, input_dtype, M, N, K, batch_size, backend):
         device = "cuda"
         dtype = input_dtype
+        prev_backend = torch.backends.cuda.preferred_blas_library()
```

Comment:
consider using a simple context manager so that backend gets restored in case of error
```
@contextmanager
def blas_library_context(backend):
    prev_backend = torch.backends.cuda.preferred_blas_library()
    torch.backends.cuda.preferred_blas_library(backend)
    try:
        yield
    finally:
        torch.backends.cuda.preferred_blas_library(prev_backend)
 ```